### PR TITLE
feat: additional compiler flags

### PR DESCRIPTION
**Pull request by fabian1409**

### CODE REVIEW

**Comment by fabian1409** [comment]
fixes (#112)

File: book/src/config.md [hunk 1]
```diff
@@ -71,6 +71,12 @@ This section configures the co-circom MPC compiler.
 * `version`: Allows leaking of secret values in loops (default: "2.0.0")
 * `link_library`: A list of strings that represent paths for circom to look for library files (`-l` flag in circom) (default: `[]`)
 * `allow_leaky_loops`: Allows leaking of secret values in loops (default: `false`) (*currently not implemented*).
+* `simplification`: Controls the simplification level applied during constraint generation (default: `O3(usize::MAX)`)
```
Comment (dkales) [inline book/src/config.md:74]: this should be O2?

File: circom-mpc-compiler/src/lib.rs [hunk 2]
```diff
@@ -42,16 +42,44 @@ use itertools::Itertools;
 use serde::{Deserialize, Serialize};
 use std::{collections::HashMap, marker::PhantomData, path::PathBuf};
 
+/// The simplification level applied during constraint generation
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, Eq, PartialEq, PartialOrd, Ord, Hash)]
+pub enum Simplification {
```
Comment (dkales) [inline circom-mpc-compiler/src/lib.rs:47]: Probably `SimplificationLevel` is a better name

File: circom-mpc-compiler/src/lib.rs [hunk 3]
```diff
@@ -142,15 +173,19 @@ where
...
+                r
+            } else {
+                0
+            },
             flag_json_sub: false,
             json_substitutions: String::new(),
-            flag_s: false,
-            flag_f: false,
+            flag_s: self.config.simplification == Simplification::O1,
+            flag_f: self.config.simplification == Simplification::O0,
```
Comment (dkales) [inline circom-mpc-compiler/src/lib.rs:184]: Are these really exclusive? What happens if both are `true`?

**Review by dkales — CHANGES_REQUESTED** [review]

File: circom-mpc-compiler/src/lib.rs (hunk 3 above)
Comment (fabian1409) [reply circom-mpc-compiler/src/lib.rs:184]: They are exclusive in circom (https://github.com/iden3/circom/blob/master/circom/src/input_user.rs#L100C1-L102C77)
If both are passed as args in circom, the "lower" (in this case 0) one will be be used, for co-cricom only one can be set in the config

**Review by dkales — APPROVED** [review]

**Pull request merged**